app/controllers/charges_controller.rb

```diff
@@ -1,25 +1,37 @@
 class ChargesController < ApplicationController
   def new
+    order = Order.find_by(id: session[:checkout_order_id])
```

Comment:
I'm still not convinced the session variable is necessary, but I can't tell just from this PR. We'll take a look at it later today.